src/main/java/com/conveyal/datatools/manager/controllers/api/ServerController.java

```diff
@@ -0,0 +1,343 @@
+import com.amazonaws.services.elasticloadbalancingv2.model.AmazonElasticLoadBalancingException;
+import com.amazonaws.services.elasticloadbalancingv2.model.DescribeTargetGroupsRequest;
+import com.amazonaws.services.elasticloadbalancingv2.model.TargetGroup;
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagement;
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagementClientBuilder;
+import com.amazonaws.services.identitymanagement.model.InstanceProfile;
+import com.amazonaws.services.identitymanagement.model.ListInstanceProfilesResult;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import com.conveyal.datatools.manager.auth.Auth0UserProfile;
+import com.conveyal.datatools.manager.models.Deployment;
+import com.conveyal.datatools.manager.models.JsonViews;
+import com.conveyal.datatools.manager.models.OtpServer;
+import com.conveyal.datatools.manager.models.Project;
+import com.conveyal.datatools.manager.persistence.FeedStore;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import com.conveyal.datatools.manager.utils.json.JsonManager;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.jetty.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.HaltException;
+import spark.Request;
+import spark.Response;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.conveyal.datatools.common.utils.SparkUtils.logMessageAndHalt;
+import static com.conveyal.datatools.manager.jobs.DeployJob.DEFAULT_INSTANCE_TYPE;
+import static spark.Spark.delete;
+import static spark.Spark.get;
+import static spark.Spark.options;
+import static spark.Spark.post;
+import static spark.Spark.put;
+
+/**
+ * Handlers for HTTP API requests that affect deployment Servers.
+ * These methods are mapped to API endpoints by Spark.
+ */
+public class ServerController {
+    private static JsonManager<OtpServer> json = new JsonManager<>(OtpServer.class, JsonViews.UserInterface.class);
+    private static final Logger LOG = LoggerFactory.getLogger(ServerController.class);
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private static final AmazonEC2 ec2 = AmazonEC2Client.builder().build();
+    private static final AmazonIdentityManagement iam = AmazonIdentityManagementClientBuilder.defaultClient();
+    private static final AmazonElasticLoadBalancing elb = AmazonElasticLoadBalancingClient.builder().build();
+
+    /**
+     * Gets the server specified by the request's id parameter and ensure that user has access to the
+     * deployment. If the user does not have permission the Spark request is halted with an error.
+     */
+    private static OtpServer checkServerPermissions(Request req, Response res) {
```

Comment:
I think this method should be renamed to `getServerWithPermissions` since it returns something.